source/modulo_components/include/modulo_components/ComponentInterface.hpp

```diff
@@ -557,6 +581,38 @@ inline bool ComponentInterface<NodeT>::get_predicate(const std::string& predicat
   return value;
 }
 
+template<class NodeT>
+inline void ComponentInterface<NodeT>::add_trigger(const std::string& trigger_name) {
+  if (trigger_name.empty()) {
+    RCLCPP_ERROR(this->get_logger(), "Failed to add trigger: Provide a non empty string as a name.");
+    return;
+  }
+  if (this->triggers_.find(trigger_name) != this->triggers_.end()
+      || this->predicates_.find(trigger_name) != this->predicates_.end()) {
+    RCLCPP_ERROR_STREAM(this->get_logger(), "Failed to add trigger: there is already a trigger or "
+                                            "predicate with name '" << trigger_name << "'.");
+    return;
+  }
+  this->triggers_.insert_or_assign(trigger_name, false);
+  this->add_predicate(
+      trigger_name, [this, &trigger_name] {
+        auto value = this->triggers_.at(trigger_name);
+        this->triggers_.at(trigger_name) = false;
+        return value;
+      }
+  );
+}
+
+template<class NodeT>
+inline void ComponentInterface<NodeT>::trigger(const std::string& trigger_name) {
+  if (this->triggers_.find(trigger_name) == this->triggers_.end()) {
+    RCLCPP_ERROR_STREAM(this->get_logger(), "Failed to trigger: could not find trigger"
+                                            " with name  '" << trigger_name << "'.");
+    return;
+  }
+  this->triggers_.at(trigger_name) = true;
```

Comment:
I think this would warrant triggering the publication, as it is in `set_predicate`
```suggestion
  this->triggers_.at(trigger_name) = true;
  publish_predicate(trigger_name);
```